src/DynamoForgeDataExchange/DynamoForgeDataExchange.cs

```diff
@@ -0,0 +1,333 @@
+            // filtering on these attributes (query based on a custom attribute not currently
+            // supported by the FDX API, but it may become available in the future)
+            var clientIdAttribute = new Attribute("clientId", "Dynamo");
+            var clientVersionAttribute = new Attribute("clientVersion", "<dynamo version>");
+            var attributes = new List<Attribute>
+            {
+                clientIdAttribute,
+                clientVersionAttribute
+            };
+
+            var exchangeComponent = new ExchangeComponent(attributes);
+            string bodyJSON = JsonConvert.SerializeObject(exchangeComponent);
+            return bodyJSON;
+        }
+
+        static public string StartFullfillment(RestClient client, string collectionId, string exchangeContainerId, string token)
+        {
+            var fulfillmentUrl = $"/v1/collections/{collectionId}/exchanges/{exchangeContainerId}/fulfillments:start";
+            RestRequest startFulfillmentRequest = new RestRequest(fulfillmentUrl);
+            AddHeadersToPostRequest(startFulfillmentRequest, token);
+            var startFulfillmentResponse = client.ExecutePost(startFulfillmentRequest);
+
+            dynamic responseBody = JObject.Parse(startFulfillmentResponse.Content);
+            // We extract the fullfilment id from the response needed from the follow up API calls
+            var fulfillmentId = responseBody["id"].Value;
+
+            return fulfillmentId;
+        }
+
+        static public void EndFullFillment(RestClient client, string collectionId, string exchangeContainerId, string fulfillmentId, string token)
+        {
+            var endFulfillmentUrl = $"/v1/collections/{collectionId}/exchanges/{exchangeContainerId}/fulfillments/{fulfillmentId}:finish";
+            RestRequest endFulfillmentRequest = new RestRequest(endFulfillmentUrl);
+            AddHeadersToPostRequest(endFulfillmentRequest, token);
+            var endFulfillmentResponse = client.ExecutePost(endFulfillmentRequest);
+            dynamic endFulfillmentResponseBody = JObject.Parse(endFulfillmentResponse.Content);
+            var fulfillmentStatus = endFulfillmentResponseBody.status.Value;
+            while (fulfillmentStatus == "IN_PROGRESS")
+            {
+                var fulfillmentStatusUrl = $"/v1/collections/{collectionId}/exchanges/{exchangeContainerId}/fulfillments/{fulfillmentId}";
+                RestRequest fulfillementStatusRequest = new RestRequest(fulfillmentStatusUrl);
+                fulfillementStatusRequest.AddHeader("Authorization", $"Bearer {token}");
+                var fulfillmentStatusResponse = client.ExecuteGet(fulfillementStatusRequest);
+                dynamic fulfillmentStatusResponseBody = JObject.Parse(fulfillmentStatusResponse.Content);
+                fulfillmentStatus = fulfillmentStatusResponseBody.status.Value;
+            }
+            Console.WriteLine($"The data exchange is {fulfillmentStatus}");
```

Comment:
Remove `Console.WriteLine` probably ported from varvara's POC app.  Maybe a good idea to log only the important ones to the Dynamo logger instead.

Reply:
Addressed this.